addons/mail/controllers/thread.py

```diff
@@ -188,9 +188,8 @@ def mail_message_post(self, thread_model, thread_id, post_data, context=None, **
         if not thread:
             raise NotFound()
         if not self._get_thread_with_access(thread_model, thread_id, mode="write"):
-            thread.env.context = frozendict(
-                thread.env.context, mail_post_autofollow_author_skip=True, mail_post_autofollow=False
-            )
+            request.update_context(mail_create_nosubscribe=True, mail_post_autofollow=False)
+            thread = thread.with_env(request.env)
```

Comment:
Can you please confirm that the existing code was intentionally modifying `request`'s context?
If not, then you shouldn't.